droidlet/interpreter/robot/tasks.py

```diff
@@ -620,3 +624,167 @@ def step(self):
             self.agent.mover.explore(self.goal)
             self.finished = self.agent.mover.nav.is_done_exploring().value
 
+class TimedExplore(TrajectorySaverTask):
+    """use slam to explore environemt"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.command_sent = False
+        self.agent = agent
+        self.goal = task_data.get("goal")
+        self.timeout = task_data.get("timeout")
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        if not self.finished and self.timeout > 0:
+            self.agent.mover.explore(self.goal)
+            self.timeout -= 1
+            self.finished = self.agent.mover.nav.is_done_exploring().value
+        else:
+            self.finished = True
+
+
+class Reexplore(Task):
+    """use slam to explore environemt, but also examine detections"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.steps = ["not_started"] * 5 # S1, C1s, C1l, R1, R2
+        self.task_data = task_data
+        self.target = task_data.get('target')
+        self.spawn_pos = task_data.get('spawn_pos')
+        self.base_pos = task_data.get('base_pos')
+        self.init_logger()
+        self.agent = agent
+        print(f'CuriousExplore task_data {task_data}')
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    def init_logger(self):
+        logger = logging.getLogger('reexplore')
+        logger.setLevel(logging.INFO)
+        fh = logging.FileHandler(f"reexplore.log", 'w')
+        fh.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(filename)s:%(lineno)s - %(funcName)s(): %(message)s')
+        fh.setFormatter(formatter)
+        logger.addHandler(fh)
+        logger.addHandler(ch)
+        logger.info(f'Reexplore task_data {self.task_data}')
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        logger = logging.getLogger('reexplore')
+
+        # execute a straigtline examine
+        if self.steps[0] == "not_started":
+            self.agent.mover.bot.respawn_agent(self.spawn_pos['position'], self.spawn_pos['rotation'])
+            base_pos = self.agent.mover.get_base_pos()
+            assert np.allclose(base_pos, self.base_pos) # checking that poses match
+
+            self.add_child_task(ExamineDetectionStraightline(
+                    self.agent, {
+                        "target": self.target, 
+                        "save_data": True,
+                        "root_data_path": f"{self.task_data['data_path']}/s1",
```

Comment:
why `root_data_path` and `data_path` instead of just one? they seem to be the exact same value

Reply:
renamed `root_data_path` to `vis_path`, a path to save visualizations which is effectively what it is used for.